RepairShop-Backend/src/test/java/ca/mcgill/ecse321/repairshop/service/TestAdminService.java

```diff
@@ -1,18 +1,536 @@
+		}
+	}
+	
+    /////////////////////////////////////////////////////////////////////////////////////////
+	
+	////////////////////////// CHANGE PHONE TEST  ///////////////////////////////////////////
+	
+	@Test
+	public void testChangePhone() {
+	
+		String newPhone = "9879879877";
+		AdminDto admin = null;
+		try {
+			admin = adminService.changePhone(ADMIN_EMAIL, newPhone);
+
+		} catch (Exception e) { 
+			// Check that no error occurred
+			fail(e.getMessage());
+		}
+		
+		assertNotNull(admin);
+		assertEquals(newPhone, admin.getPhoneNumber());
+		assertEquals(ADMIN_EMAIL, admin.getEmail());
+		
+	}
+	
+	
+	
+	@Test
+	public void testChangePhoneNull() {
+				
+		try {
+			adminService.changePhone(null, null);
+			fail();
+		} catch (Exception e) {
+			//error should occur
+			assertEquals("Email or new phone cannot be empty.", e.getMessage());
+		}
+	}
+	
+	
+	@Test
+	public void testChangePhoneNonExistentAdmin() {
+				
+		try {
+			adminService.changePhone("nobody's phone", "somebody's phone");
+			fail();
+		} catch (Exception e) {
+			//error should occur
+			assertEquals("Admin not found.", e.getMessage());
+		}
+	}
+	
+    /////////////////////////////////////////////////////////////////////////////////////////
+	
+	////////////////////////// CHANGE EMAIL TEST  ///////////////////////////////////////////
+	
+	@Test
+	public void testChangeEmail() {
+	
+		String newEmail = "gotNewColtraneMail@gmail.com";
+		AdminDto admin = null;
+		try {
+			admin = adminService.changeEmail(ADMIN_EMAIL, newEmail);
+
+		} catch (Exception e) { 
+			// Check that no error occurred
+			fail(e.getMessage());
+		}
+		
+		assertNotNull(admin);
+		assertEquals(newEmail, admin.getEmail());
+		assertNotEquals(ADMIN_EMAIL, admin.getEmail());
+		
+	}
+	
+	
+	
+	@Test
+	public void testChangeEmailNull() {
+				
+		try {
+			adminService.changeEmail(null, null);
+			fail();
+		} catch (Exception e) {
+			//error should occur
+			assertEquals("Email or new email cannot be empty.", e.getMessage());
+		}
+	}
+	
+	
+	@Test
+	public void testChangeEmailNonExistentAdmin() {
+				
+		try {
+			adminService.changeEmail("nobody@mail.com", "somebody@mail.com");
+			fail();
+		} catch (Exception e) {
+			//error should occur
+			assertEquals("Admin not found.", e.getMessage());
+		}
+	}
+	
+    /////////////////////////////////////////////////////////////////////////////////////////
+	
+	////////////////////////// CHANGE NAME TEST  ////////////////////////////////////////////
+	
+	@Test
+	public void testChangeName() {
+	
+		String newName = "Alice Coltrane";
+		AdminDto admin = null;
+		try {
+			admin = adminService.changeName(ADMIN_EMAIL, newName);
+
+		} catch (Exception e) { 
+			// Check that no error occurred
+			fail(e.getMessage());
+		}
+		
+		assertNotNull(admin);
+		assertEquals(newName, admin.getName());
+		assertNotEquals(ADMIN_NAME, admin.getName());
+		
+	}
+	
+	
+	
+	@Test
+	public void testChangeNamelNull() {
```

Comment:
Not really important, but there's an extra "l" between Name & Null

```suggestion
	public void testChangeNameNull() {
```

Reply:
This has been resolved by https://github.com/McGill-ECSE321-Winter2021/project-group-09/pull/124/commits/2b4aeae8baa1dc9bbcbe185b50a1d075c1a383ae